File: src/main/java/de/maggiwuerze/xdccloader/persistence/TargetBotRepository.java
package de.maggiwuerze.xdccloader.persistence;

import de.maggiwuerze.xdccloader.model.entity.Bot;
import java.util.List;
import org.springframework.data.repository.PagingAndSortingRepository;
import org.springframework.stereotype.Repository;


@Repository
public interface TargetBotRepository extends PagingAndSortingRepository<Bot, Long> {

	List<Bot> findAll();
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/persistence/TargetBotRepository.kt
package de.maggiwuerze.xdccwebloader.persistence

import de.maggiwuerze.xdccwebloader.persistence.entity.Bot
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface TargetBotRepository : CrudRepository<Bot, UUID> {
    override fun findAll(): List<Bot>

    fun findByName(name: String): Bot?
}